client/src/components/Header.tsx

```diff
@@ -95,7 +98,27 @@ function AccountMenu() {
           'aria-labelledby': 'menu-btn',
         }}
       >
-        <MenuItem onClick={closeMenu}>My Posts</MenuItem>
+        <MenuItem
+          onClick={async () => {
+            try {
+              const result = await api.me();
+              const user = result.data;
+              if (!user) {
+                throw new Error('Current user could not be found.');
+              }
+
+              const query = `${user.firstName} ${user.lastName}`;
+              props.setSearch(query);
```

Comment:
What happens if a user is mentioned in a post or comment? Does that return for `My Posts`? IMO there should be a myPosts endpoint on the API for this functionality.